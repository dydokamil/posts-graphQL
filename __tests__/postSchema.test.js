jest.mock('axios')

const { graphql } = require('graphql')
const schema = require('../schema')

describe('post schema', () => {
  it('should fetch a list of posts', async () => {
    const query = `
      {
        posts {
          id
          author {
            id
            username
            email
            createdAt
            lastLogin
            posts {
              id 
              createdAt 
              editedAt
              message
            }
          }
          createdAt
          editedAt
          message
        }
      } 
    `

    const result = await graphql(schema, query)
    const posts = result.data.posts
    expect(posts.length).toBe(5)
    expect(posts).toMatchSnapshot()
  })

  it('should respond with a single post', async () => {
    const id = 4
    const query = `
      {
        post(postId: ${id}) {
          id
          author {
            id
            username
            email
            createdAt
            lastLogin
            posts {
              id 
              createdAt 
              editedAt
              message
            }
          }
          createdAt
          editedAt
          message
        }
      } 
    `

    const result = await graphql(schema, query)
    const post = result.data.post
    expect(post).toMatchSnapshot()
  })
})
