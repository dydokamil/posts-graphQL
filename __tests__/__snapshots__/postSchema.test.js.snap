// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post schema should fetch a list of posts 1`] = `
Array [
  Object {
    "author": Object {
      "createdAt": null,
      "email": null,
      "id": null,
      "lastLogin": null,
      "posts": null,
      "username": null,
    },
    "createdAt": "2018-10-10T13:00:00",
    "editedAt": "2018-10-10T13:00:00",
    "id": "1",
    "message": "Some message",
  },
  Object {
    "author": Object {
      "createdAt": null,
      "email": null,
      "id": null,
      "lastLogin": null,
      "posts": null,
      "username": null,
    },
    "createdAt": "2018-10-10T13:00:00",
    "editedAt": "2018-10-10T13:00:00",
    "id": "2",
    "message": "Some message",
  },
  Object {
    "author": Object {
      "createdAt": null,
      "email": null,
      "id": null,
      "lastLogin": null,
      "posts": null,
      "username": null,
    },
    "createdAt": "2018-10-10T13:00:00",
    "editedAt": "2018-10-10T13:00:00",
    "id": "3",
    "message": "Some message",
  },
  Object {
    "author": Object {
      "createdAt": null,
      "email": null,
      "id": null,
      "lastLogin": null,
      "posts": null,
      "username": null,
    },
    "createdAt": "2018-10-10T13:00:00",
    "editedAt": "2018-10-10T13:00:00",
    "id": "4",
    "message": "Some message",
  },
  Object {
    "author": Object {
      "createdAt": null,
      "email": null,
      "id": null,
      "lastLogin": null,
      "posts": null,
      "username": null,
    },
    "createdAt": "2018-10-10T13:00:00",
    "editedAt": "2018-10-10T13:00:00",
    "id": "5",
    "message": "Some message",
  },
]
`;

exports[`post schema should respond with a single post 1`] = `
Object {
  "author": Object {
    "createdAt": null,
    "email": null,
    "id": null,
    "lastLogin": null,
    "posts": null,
    "username": null,
  },
  "createdAt": "2018-10-10T13:00:00",
  "editedAt": "2018-10-10T13:00:00",
  "id": "4",
  "message": "Some message",
}
`;
